### PR TITLE
Implement RFC 57: Single-Field Register Definition and Usage Shortcut

### PR DESCRIPTION
**Pull request by tpwrules**
Tested on my design and it works nice.

### CODE REVIEW

File: amaranth_soc/csr/reg.py [hunk 1]
```diff
@@ -562,18 +564,24 @@ def __iter__(self):
...
 
-        for field_path, field in self.fields.flatten():
+        for field_path, field in self:
             field_width = Shape.cast(field.port.shape).width
             field_slice = slice(field_start, field_start + field_width)
 
-            m.submodules["__".join(str(key) for key in field_path)] = field
+            # avoid empty name for a single un-named field
+            name = "__".join(str(key) for key in field_path)
+            if name == "": name = "action"
```
Comment (jfng) [inline amaranth_soc/csr/reg.py:583]: I'd prefer the following, which differentiates against an edge-case (a field named "action"):

```python3
if field_path:
    m.submodules["__".join(str(key) for key in field_path)] = field
else:
    m.submodules += field
```

**Review by jfng — CHANGES_REQUESTED** [review]
LGTM besides one nitpick.

**Comment by tpwrules** [comment]
Seems reasonable, fixed.

**Review by jfng — APPROVED** [review]

**Comment by jfng** [comment]
Thank you!

**Pull request merged**